lib/src/chunked_stream_reader.dart

```diff
@@ -97,29 +98,38 @@ class ChunkedStreamReader<T> {
       // While we have data to read
       while (size > 0) {
         // Read something into the buffer, if it's empty
-        if (_buffer.isEmpty) {
+        if (_buffer.length - _offset <= 0) {
           if (!(await _input.moveNext())) {
             // Don't attempt to read more data, as there is no more data.
             size = 0;
             _reading = false;
             break;
           }
           _buffer = _input.current;
+          _offset = 0;
         }
 
-        if (_buffer.isNotEmpty) {
-          if (size < _buffer.length) {
-            final output = _buffer.sublist(0, size);
-            _buffer = _buffer.sublist(size);
+        if (_buffer.length - _offset > 0) {
+          if (size < _buffer.length - _offset) {
+            late List<T> output;
+            if (_buffer is Uint8List) {
```

Comment:
That's why we have `ChunkedStreamReaderByteStreamExt`, but yes, in hindsight... maybe we should only have made `ChunkedByteStreamReader` -- because in practice most other streams aren't chunked.